### PR TITLE
feat(aws-eventbridge-sns , aws-events-rule-sns, aws-events-rule-lambda): custom event bus support

### PR DESCRIPTION
**Pull request by surukonda**
*Issue #, if available:*

*Description of changes:*

aws-eventbridge-sns - Custom EventBus support
aws-events-rule-sns - Custom EventBus support
aws-events-rule-lambda - Custom EventBus support
aws-eventbridge-lambda - used semicolon(;) consistently in interface declaration
core/eventbridge-helper.ts -  used semicolon(;) consistently in interface declaration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 05f093f7a020b985adb06bc588ad3fedb008a91e
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=21452ad5-dd33-4a12-a3ce-9fde56dc8803%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: f9d493784ac24a671ae9bda54bc6a6727e7cfc1e
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=970ae514-b75e-4b56-8fd1-5a9f1d06cd92%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — CHANGES_REQUESTED** [review]

File: source/patterns/@aws-solutions-constructs/aws-eventbridge-sns/test/eventbridge-sns-topic.test.ts [hunk 1]
```diff
@@ -261,4 +273,100 @@ test('check the sns topic properties with existing KMS key', () => {
     Description: "my-key",
     EnableKeyRotation: true
   });
+});
+
+test('check eventbus property', () => {
+  const stack = new cdk.Stack();
+
+  const construct: EventbridgeToSns = deployStackWithNewEventBus(stack);
```
[the remark this answers is not in the document]
Comment (surukonda) [reply source/patterns/@aws-solutions-constructs/aws-eventbridge-sns/test/eventbridge-sns-topic.test.ts:281]: makes sense, will update it

File: source/patterns/@aws-solutions-constructs/aws-events-rule-lambda/test/events-rule-lambda.test.ts [hunk 2]
```diff
@@ -33,6 +33,23 @@ function deployNewFunc(stack: cdk.Stack) {
   return new EventsRuleToLambda(stack, 'test-events-rule-lambda', props);
 }
 
+function deployNewEventBus(stack: cdk.Stack) {
```
[the remark this answers is not in the document]
Comment (surukonda) [reply source/patterns/@aws-solutions-constructs/aws-events-rule-lambda/test/events-rule-lambda.test.ts:36]: will update the PR

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 255aee85269cfbf40d071f70e52b7669728b14f9
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=65cafa67-7eeb-452b-85bf-668c46f07430%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Pull request merged**